Add `stm32_min_dev` support to `wkup_pins` example.

arduino12 opened the pull request:
Summery:
3mA with PWR LED.
24uA after disableing PWR LED (remove R1).
58nA after removing 3V regulator (remove U1).

PU on WAKEUP pin doesn't work on deepsleep, need to use external PU!
Cant program when MCU is sleeping - need to reset / wakeup first.
External light hitting debug LED cause ~100uA noise!
Temperature affects the current draw.

arduino12 committed:
samples: boards: st: power_mgmt: wkup_pins: add `stm32_min_dev` overlay

Current consumption at 3.3V:
3mA with PWR LED.
24uA after disabling PWR LED (remove R1).
58nA after removing 3V regulator (remove U1).
Notes:
PU on WAKEUP pin doesn't work on deep-sleep, need to use an external pull-up resistor!
External light hitting debug LED cause ~100uA noise!
Temperature affects the current draw.

Signed-off-by: Arad Eizen <arad.rgb@gmail.com>

arduino12 committed:
samples: boards: st: power_mgmt: wkup_pins: add `stm32_min_dev` overlay

samples: boards: st: power_mgmt: wkup_pins: add `stm32_min_dev` overlay

Current consumption at 3.3V:
3mA with PWR LED.
24uA after disabling PWR LED (remove R1).
58nA after removing 3V regulator (remove U1).
Notes:
PU on WAKEUP pin doesn't work on deep-sleep, need to use an external pull-up resistor!
External light hitting debug LED cause ~100uA noise!
Temperature affects the current draw.

Signed-off-by: Arad Eizen <arad.rgb@gmail.com>

etienne-lms reviewed (commented):
Could you refine the commit message: header line should start with something like:
`samples: boards: st: power_mgmt: wkup_pins: ...` and `Signed-off-by` tag is missing.

arduino12 commented on samples/boards/st/power_mgmt/wkup_pins/boards/stm32_min_dev.overlay:
```diff
@@ -0,0 +1,30 @@
+
+		user_button: button_0 {
+			label = "User SW0";
+			gpios = <&gpioa 0 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
```
This is the only pin supporting wakeup from "standby mode" (deep-sleep) its even called `PA0-WKUP`.
Thanks for all your CR - I will fix everithing soon!

etienne-lms reviewed (commented):
Changes LGTM but could you split the changes in 2 commits? one for the stm32_min board overlay file changes and one for the cleanup changes in the 2 other boards overlay files?

The commit message header line is duplicated, could you fix that?

arduino12 committed:
samples: boards: st: power_mgmt: wkup_pins: remove key code in overlays

Remove zephyr,code properties in DTS overlay file that are not needed.
By the way, fix an inline comment capitalization typo.
No functional change.

Signed-off-by: Arad Eizen <arad.rgb@gmail.com>

arduino12 committed:
Merge branch 'main' of https://github.com/arduino12/zephyr

arduino12 committed:
samples: boards: st: power_mgmt: stop3: add CONFIG_PINCTRL to prj.conf

Build failed with:
zephyr/include/zephyr/drivers/pinctrl.h:351: undefined reference to `pinctrl_lookup_state'
zephyr/include/zephyr/drivers/pinctrl.h:332: undefined reference to `pinctrl_configure_pins'
Adding `CONFIG_PINCTRL=y` to prj.conf fixed that.

Signed-off-by: Arad Eizen <arad.rgb@gmail.com>
